packages/contracts-bedrock/src/L1/DataAvailabilityChallenge.sol

```diff
@@ -0,0 +1,250 @@
+    /// @param status The new status of the challenge.
+    event ChallengeStatusChanged(
+        bytes32 indexed challengedHash, uint256 indexed challengedBlockNumber, ChallengeStatus status
+    );
+
+    /// @notice An event that is emitted when the head of the chain of expired challenges is updated.
+    /// @param expiredChallengesHead The new head of the chain of expired challenges.
+    event ExpiredChallengesHeadUpdated(bytes32 expiredChallengesHead);
+
+    /// @notice Semantic version.
+    /// @custom:semver 0.0.0
+    string public constant version = "0.0.0";
+
+    /// @notice The block interval during which a commitment can be challenged.
+    uint256 public challengeWindow;
+
+    /// @notice The block interval during which a challenge can be resolved.
+    uint256 public resolveWindow;
+
+    /// @notice The amount required to post a challenge.
+    uint256 public bondSize;
+
+    /// @notice A mapping from addresses to their bond balance in the contract.
+    mapping(address => uint256) public balances;
+
+    /// @notice A mapping from challenged block numbers to challenged hashes to challenges.
+    mapping(uint256 => mapping(bytes32 => Challenge)) public challenges;
+
+    /// @notice The head of the chain of expired challenges.
+    bytes32 public expiredChallengesHead;
+
+    /// @notice constructs a new DataAvailabilityChallenge contract.
+    constructor() OwnableUpgradeable() {}
+
+    /// @notice Sets the challenge window.
+    /// @param _challengeWindow The block interval during which a commitment can be challenged.
+    function setChallengeWindow(uint256 _challengeWindow) public onlyOwner {
+        challengeWindow = _challengeWindow;
+    }
+
+    /// @notice Sets the resolve window.
+    /// @param _resolveWindow The block interval during which a challenge can be resolved.
+    function setResolveWindow(uint256 _resolveWindow) public onlyOwner {
+        resolveWindow = _resolveWindow;
+    }
+
+    /// @notice Sets the bond size.
+    /// @param _bondSize The amount required to post a challenge.
+    function setBondSize(uint256 _bondSize) public onlyOwner {
+        bondSize = _bondSize;
+    }
+
+    /// @notice Initializes the contract.
+    /// @param _owner The owner of the contract.
+    /// @param _challengeWindow The block interval during which a commitment can be challenged.
+    /// @param _resolveWindow The block interval during which a challenge can be resolved.
+    /// @param _bondSize The amount required to post a challenge.
+    function initialize(address _owner, uint256 _challengeWindow, uint256 _resolveWindow, uint256 _bondSize) public initializer {
+        __Ownable_init();
+        setChallengeWindow(_challengeWindow);
+        setResolveWindow(_resolveWindow);
+        setBondSize(_bondSize);
+        _transferOwnership(_owner);
+    }
+
+    /// @notice Post a bond as prerequisite for challenging a commitment.
+    function deposit() external payable {
```

Comment:
Can you add a receive function that calls `deposit`? So you can just send ether directly to the contract without needing an ABI